src/kernels/raw/launcher/kernelProcess.node.ts

```diff
@@ -292,6 +300,35 @@ export class KernelProcess implements IKernelProcess {
         });
     }
 
+    private async killChildProcesses(pid?: number) {
+        // Do not remove this code, in in unit tests we end up running this,
+        // then we run into the danger of kill all of the processes on the machine.
+        // because calling `pidtree` without a pid will return all pids and hence everything ends up getting killed.
+        if (!pid || !ProcessService.isAlive(pid)) {
+            return;
+        }
+        try {
+            if (this.platform.isWindows) {
+                const windir = process.env['WINDIR'] || 'C:\\Windows';
+                const TASK_KILL = path.join(windir, 'System32', 'taskkill.exe');
+                await new ProcessService().exec(TASK_KILL, ['/F', '/T', '/PID', pid.toString()]);
+            } else {
+                await new Promise<void>((resolve) => {
+                    pidtree(pid, (ex: unknown, pids: number[]) => {
```

Comment:
>  rather than waiting the full time it takes to kill them?

If you check carefully, we dont wait to kill them, we wait until we've got the full list. 
I cannot return before we get the list, as the next line of code is to kill the parent process, and that cannot happen too early, else i'll never get the list of the child procs